components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -1019,15 +1011,18 @@ public AccessTokenDO getAccessToken(String accessTokenIdentifier, boolean includ
                     String subjectIdentifier = resultSet.getString(14);
                     String authenticatedIDP = null;
                     String tokenBindingReference = resultSet.getString(15);
-                    if (OAuth2ServiceComponentHolder.isIDPIdColumnEnabled()) {
-                        authenticatedIDP = resultSet.getString(16);
-                    }
-
+                    authenticatedIDP = resultSet.getString(16);
                     boolean isConsentedToken = false;
-                    if (isConsentedColumnDataFetched) {
+                    if (!includeExpired && OAuth2ServiceComponentHolder.isConsentedTokenColumnEnabled()) {
```

Comment:
what is reason to add `!includeExpired` here?

Reply:
Here I have combined the above if conditions (L967 - L987) with this once since there were lots of if else conditions.

Reply:
Yes but `!includeExpired` is not a check in the original logic right?

Reply:
ack. however we don't need this check at L1023